src/fetch-map/types.ts

```diff
@@ -205,3 +207,63 @@ export type GoogleBasemapProps = {
   tilt?: number;
   heading?: number;
 };
+
+export type Dataset = {
+  id: string;
+  type: MapType;
+  source: string;
+  cache?: number;
+  connectionName: string;
+  geoColumn: string;
+  data: TilejsonResult | GeojsonResult | JsonResult;
+  columns: string[];
+  format: Format;
+  aggregationExp: string;
+  aggregationResLevel: number;
+  queryParameters: QueryParameters;
+};
+
+export enum AttributeType {
+  String = 'String',
+  Number = 'Number',
+  Timestamp = 'Timestamp',
+  Boolean = 'Boolean',
+}
```

Comment:
Minor – I'm mostly trying to use type unions (`'String' | 'Number' | ...`) for public-facing code, for consistency in the API, would that be OK here? Have made some exceptions, particularly for cases where the underlying string values are 'weird'.